### ISSUE TITLE
Is there a GPU version?

### ISSUE THREAD

**Issue by Caoxuheng** [yingutk/u2MDN#1]
Thanks for your work on u2MDN.When I ran this code on the GPU, some errors occurred. Running it on the CPU was successful, but it took too much time. Is there a GPU version?

**Comment by yingutk** [comment]
Hi,

This should be able to run on a GPU version. I wonder if the problem is caused by the version of tensor flow.

What is the problem when you run it on a GPU?

Best,
Ying


On Apr 19, 2023, at 15:07, Caoxuheng ***@***.***> wrote:



Thanks for your work on u2MDN.When I ran this code on the GPU, some errors occurred. Running it on the CPU was successful, but it took too much time. Is there a GPU version?

**Comment by Caoxuheng** [comment]
Hello Ying, thank you very much for your reply.  
I have checked the environment configuration and they meet the requirements.  This is the version information of the Tensorflow: tensorboard          1.13.1
tensorflow           1.13.1
tensorflow-estimator 1.13.0  
tensorflow gpu avaliable is True; Gpu information: 24G RTX3090;  
 
Also, the error information is pasted here. I searched for some plans and tried to solve this problem, but nothing ahead. 
 tensorflow/stream_executor/cuda/cuda_blas.cc:698] failed to run cuBLAS routine cublasSgemm_v2: CUBLAS_STATUS_EXECUTION_FAILED
InternalError (see above for traceback): Blas GEMM launch failed : a.shape=(1048576, 3), b.shape=(3, 3), m=1048576, n=3, k=3
	 [[node hsi_encoder_1/fully_connected/Tensordot/MatMul (defined at D:\Spectral super_resolution\u2MDN-main\pbAuto_mi.py:118) ]]
	 [[node concat_27 (defined at D:\Spectral super_resolution\u2MDN-main\pbAuto_mi.py:96) ]]  
pbAuto_mi.py:118--->layer_11 = tf.contrib.layers.fully_connected(x, self.nLRlevel[0], activation_fn=None)  
pbAuto_mi.py:96---> stick_segment_sum_lr = tf.concat([stick_segment_sum_lr, tf.expand_dims(stick_segment, -1)],-1)

This may caused by the version of cudnn?
